rcl_logging_interface/include/rcl_logging_interface/rcl_logging_interface.h

```diff
@@ -91,6 +95,59 @@ RCL_LOGGING_INTERFACE_PUBLIC
 RCUTILS_WARN_UNUSED
 rcl_logging_ret_t rcl_logging_external_set_logger_level(const char * name, int level);
 
+/// Get the logging directory.
+/**
+ * Uses various environment variables to construct a logging directory path.
+ *
+ * Use $ROS2_LOG_DIR if ROS2_LOG_DIR is set and not empty.
+ * Otherwise, use $ROS2_HOME/log, using ~/.ros for ROS2_HOME if not set or if empty.
+ *
+ * It also expands '~' to the current user's home directory.
+ *
+ * \param[inout] directory The C string pointer at which to write the directory path.
+ *   Only meaningful if the call is successful.
+ * \param[in] allocator The allocator to use for memory allocation.
+ * \return RCL_LOGGING_RET_OK if successful, or
+ * \return RCL_LOGGING_RET_ERROR if an unspecified error occurs.
+ */
+RCL_LOGGING_INTERFACE_PUBLIC
+inline
```

Comment:
(feel free to ignore this comment)

I don't mind the inline implementation, in that way `rcl_logging_interface` continues being header only.
But I would rather put the implementation in a separate file, e.g.: `impl/rcl_logging_interface.inc`.
You can `#include <rcl_logging_interface/impl/rcl_logging_interface.inc` at the end of this file.

Reply:
I would prefer to avoid the inline myself... Maybe this shouldn't be in `rcl_logging_interface`? Or maybe it should just have a library...

`rmw`, by comparison, has a library `librmw` as well as providing headers which must be implemented by rmw implementations. So I think this could follow a similar pattern.

Reply:
> I would prefer to avoid the inline myself... Maybe this shouldn't be in rcl_logging_interface? 

I was thinking about adding it to another place. The only place that sounded reasonable to me was `rcl`, but in that case we cannot use the function here because it would create a circular dependecy.

> Or maybe it should just have a library...

sounds good to me

Reply:
changed in 35cc88ad261cd40d97f5d3d1e4c2edba78a8d69b. I *think* I did it correctly.